From ca1cdaaff0a899fce471bccd3e52400d9f9edd16 Mon Sep 17 00:00:00 2001
From: Vadim-Valdis Yudaev <judaew@outlook.de>
Date: Sat, 6 Nov 2021 22:24:19 +0200
Subject: [PATCH] Use XDG Base Directory specification

I think the best idea is to store non-essential files (that can be
deleted without harm) in XDG_CACHE_HOME, as required by the
XDG Base Directory specification.

See https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

For systems that do not have this environment variable by default, can
use ${HOME}/.cache.

This will help reduce the number of directories in the user's home
directory.
---
 spotify_dl/constants.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

index ab79c3f..62c93e5 100644
--- spotify_dl/constants.py.orig
+++ spotify_dl/constants.py
@@ -1,4 +1,10 @@
+import os
+
 __all__ = ['VERSION']
 
 VERSION = '7.6.0'
-SAVE_PATH = '~/.spotifydl'
+
+if os.getenv("XDG_CACHE_HOME") is not None:
+    SAVE_PATH = os.getenv("XDG_CACHE_HOME") + "/spotifydl"
+else:
+    SAVE_PATH = os.getenv("HOME") + ".cache/spotifydl"
